NBitcoin/Protocol/NodesGroup.cs

```diff
@@ -161,10 +161,12 @@ internal void StartConnecting()
 								var groupSelector = CustomGroupSelector != null ? CustomGroupSelector :
 													AllowSameGroup ? WellKnownGroupSelectors.ByRandom : null;
 								node = Node.Connect(_Network, parameters, _ConnectedNodes.Select(n => n.RemoteSocketEndpoint).ToArray(), groupSelector);
-								var timeout = CancellationTokenSource.CreateLinkedTokenSource(_Disconnect.Token);
-								timeout.CancelAfter(5000);
-								node.VersionHandshake(_Requirements, timeout.Token);
-								NodeServerTrace.Information("Node successfully connected to and handshaked");
+								using(var timeout = CancellationTokenSource.CreateLinkedTokenSource(_Disconnect.Token))
+								{
+									timeout.CancelAfter(5000);
+									node.VersionHandshake(_Requirements, timeout.Token);
+									NodeServerTrace.Information("Node successfully connected to and handshaked");
+								}

```

Comment:
This PR is for fixing a memory leak we found in Wasabi and nothing else. I don't want to fix something that is not broken and if well it could be okay (or even recommended by some people) to dispose these cts (and in general all the Disposable objects) that is not necessary in this case.

Let's put aside the recommendations, styling and other considerations and focus on the evidence here. The two images in this PR description show the heap structure before and after these changes. It is clear the problem was fixed, meaning that there are no more leaks of this type. 

Sadly, the first image is all the evidence that I have because I deleted the original heap dump. However, I've been watching the wasabi's heap behavior after these PR and I look very well (no more leaking):

[capture-1.txt](https://github.com/MetacoSA/NBitcoin/files/2465750/capture-1.txt)
[capture-2.txt](https://github.com/MetacoSA/NBitcoin/files/2465751/capture-2.txt)
[capture-3.txt](https://github.com/MetacoSA/NBitcoin/files/2465752/capture-3.txt)
[capture-4.txt](https://github.com/MetacoSA/NBitcoin/files/2465753/capture-4.txt)
[capture-5.txt](https://github.com/MetacoSA/NBitcoin/files/2465754/capture-5.txt)

Finally, I've been running the wallt with these changes for almost two days now, having a mempool with more than 500,000 transactions and it is using 297 MB RAM, what looks okay to me because the memory usage grows very very slowly, enough to keep the mempool.

Reply:
> This PR is for fixing a memory leak we found in Wasabi and nothing else. 

Fair enough. You could dispose them in a separate commit though.

Reply:
My point is, if something is asynchronously kicked off to run and the cts token is disposed before the timeout, then that's an issue.

Reply:
Oh. Ok, I will review it.

Reply:
I reviewed it and didn't find any way the cts can be disposed before time. Here we wrapped the `VersionHandshake` invocation that, if well all the communication messages are sent/received asynchronously, finishes only when the handshaking is done or failed. So, the cts is always disposed after all the async tasks are completed.